nilearn/glm/first_level/first_level.py

```diff
@@ -1321,6 +1323,94 @@ def _prepare_mask(self, run_img):
+        figsize=(10, 6),
+    ):
+        """Plot the predicted time series and residuals for a voxel \
+        or small region.
+
+        The :term:`GLM` must be fitted and have the computed design
+        matrix(ces).
+
+        Parameters
+        ----------
+        coords: tuple of coordinates
+            Coordinates of the voxel or region center.
+        masker : NiftiMasker or NiftiSpheresMasker, optional
+            Custom masker used to extract the time series. If None, a
+            :class:`~nilearn.maskers.NiftiSpheresMasker` centered on `coords`
+            with radius `radius` is created.
+        radius : float, optional
+            Radius of the sphere if `masker` is None. Default is 3mm.
+        figsize : tuple, optional
+            Size of the figure. Default is (10, 6).
+
+        Returns
+        -------
+        fig : matplotlib.figure.Figure
+            The matplotlib figure containing two subplots:
+            - Top: observed vs predicted signal
+            - Bottom: residuals
+
+        Notes
+        -----
+        This method requires that the model was fitted with
+        ``minimize_memory=False``, since the voxelwise predicted signal
+        and residuals are only stored in that mode.
+
+        """
+        check_is_fitted(self)
+
+        if self.minimize_memory:
+            raise ValueError(
+                "To plot predicted signal and residuals, "
+                "the `FirstLevelModel`-object needs to store "
+                "there attributes. "
+                "To do so, set `minimize_memory` to `False` "
+                "when initializing the `FirstLevelModel`-object."
+            )
+
+        # Create masker if needed
+        if masker is None:
+            masker = NiftiSpheresMasker([coords], radius=radius)
+            masker.fit()
+
+        # Get observed, predicted, and residual time series
+        y_pred = self._get_element_wise_model_attribute(
+            "predicted", result_as_time_series=True
+        )
+        resid = self._get_element_wise_model_attribute(
+            "residuals", result_as_time_series=True
+        )
+
+        # Extract time series for the observed, predicted, and residuals
+        predicted_ts = masker.transform(y_pred[0])
+        residuals_ts = masker.transform(resid[0])
+        observed_ts = predicted_ts + residuals_ts
+
+        # Plot observed vs predicted signal
+        fig, axes = plt.subplots(2, 1, figsize=figsize, sharex=True)
+        axes[0].plot(observed_ts, label="Observed", color="blue")
+        axes[0].plot(predicted_ts, label="Predicted", color="orange")
+        axes[0].set_title("Observed vs Predicted Signal")
+        axes[0].set_ylabel("Signal Intensity")
+        axes[0].legend()
+
+        # Plot residuals
+        axes[1].plot(residuals_ts, label="Residuals", color="red")
+        axes[1].set_title("Residuals")
+        axes[1].set_xlabel("Time")
+        axes[1].set_ylabel("Residuals")
+        axes[1].legend()
```

Comment:
make sure the plot of the residuals
- is centered on y=0
- has a horizontal line for the y=0